docs/topics/native/native-definition-file.md

```diff
@@ -0,0 +1,266 @@
+This is a property file that describes how exactly the library should be consumed. See the full [list of available properties](#properties).
+
+Here's a general workflow when working with a project:
+
+1. Create a `.def` file describing what to include into bindings.
+2. Use generated bindings in your Kotlin code.
+3. Run the Kotlin/Native compiler to produce the final executable.
+
+## Create and configure a definition file
+
+Let's create a definition file and generate bindings for a C library:
+
+1. In your IDE, select the `src` folder and create a new directory with **File | New | Directory**.
+2. Name new directory **nativeInterop/cinterop**.
+   
+   This is the default convention for `.def` file locations, though it can
+   be overridden in the `build.gradle.kts` file if you use a different location.
+3. Select this new subfolder and create a `png.def` file with **File | New | File**.
+4. Add the necessary properties:
+
+   ```none
+   headers = png.h
+   headerFilter = png.h
+   package = png
+   
+   compilerOpts.linux = -I/usr/include -I/usr/include/x86_64-linux-gnu
+   linkerOpts.osx = -L/opt/local/lib -L/usr/local/opt/png/lib -lpng
+   linkerOpts.linux = -L/usr/lib/x86_64-linux-gnu -lpng
+   ```
+
+   * `headers` is the list of header files to generate Kotlin stubs. You can add multiple files to this entry,
+     separating each with a space. In this case, it's only `png.h`. The referenced files need to be available
+     on the specified path (in this case, it's `/usr/include/png`).
+   * `headerFilter` shows what exactly is included. In C, all the headers are also included when one file references
+     another one with the `#include` directive. Sometimes it's not necessary, and you can add this parameter
+     [using glob patterns](https://en.wikipedia.org/wiki/Glob_(programming)) to fine-tune things.
+
+     You can use `headerFilter` if you don't want to fetch external dependencies (such as system `stdint.h` header) into
+     the interop library. Also, it may be useful for library size optimization and fixing potential conflicts between
+     the system and the provided Kotlin/Native compilation environment.
+
+   * If the behavior for a certain platform needs to be modified, you can use a format like `compilerOpts.osx` or
+     `compilerOpts.linux` to provide platform-specific values to the options. In this case, they are macOS
+     (the `.osx` suffix) and Linux (the `.linux` suffix). Parameters without a suffix are also possible
+     (for example, `linkerOpts=`) and applied to all platforms.
+
+5. To generate bindings, synchronize the Gradle files by clicking **Sync Now** in the notification.
+
+   ![Synchronize the Gradle files](gradle-sync.png)
+
+After the bindings generation, the IDE can use them as a proxy view of the native library.
+
+> You can also configure bindings generation by using the [`cinterop` tool](#generate-bindings-using-command-line) in the command line.
+> 
+{type="tip"}
```

Comment:
```suggestion
```
Not needed in writerside

Reply:
I kept it for consistency, but changed to "style"